src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/PackageRestore/PackageRestoreDataSource.cs

```diff
@@ -116,16 +124,44 @@ internal async Task<IEnumerable<IProjectVersionedValue<RestoreData>>> RestoreAsy
             };
         }
 
+        // means either it has never receive computed value from configured project yet, or the computed data is out of dated.
+        private void CheckPendingNomination(IProjectVersionedValue<PackageRestoreUnconfiguredInput> e)
```

Comment:
This method only uses `e.Value.ConfiguredInputs`. It might be clearer to change the argument type to a more specific type.

Reply:
Done. Passing IReadOnlyCollection<PackageRestoreConfiguredInput>